RFS/src/main/java/com/rfs/worker/DocumentsRunner.java

```diff
@@ -1,74 +1,81 @@
-            unpackerFactory,
-            reader,
-            reindexer
-        );
-    }
+    public void migrateNextShard() throws IOException {
+        workCoordinator.ensurePhaseCompletion(wc -> {
+                    try {
+                        return wc.acquireNextWorkItem(Duration.ofMinutes(10));
+                    } catch (Exception e) {
+                        throw Lombok.sneakyThrow(e);
+                    }
+                },
+                new IWorkCoordinator.WorkAcquisitionOutcomeVisitor<Void>() {
+                    @Override
+                    public Void onAlreadyCompleted() throws IOException {
+                        return null;
+                    }
 
-    @Override
-    public void runInternal() {
-        WorkerStep nextStep = null;
-        try {
-            nextStep = new DocumentsStep.EnterPhase(members);
+                    @Override
+                    public Void onAcquiredWork(IWorkCoordinator.WorkItemAndDuration workItem) throws IOException {
+                        doDocumentsMigration(IndexAndShard.valueFromWorkItemString(workItem.getWorkItemId()));
+                        return null;
+                    }
+                });
+    }
 
-            while (nextStep != null) {
-                nextStep.run();
-                nextStep = nextStep.nextStep();
-            }
-        } catch (Exception e) {
-            throw new DocumentsMigrationPhaseFailed(
-                members.globalState.getPhase(), 
-                nextStep, 
-                members.cmsEntry.map(bar -> (CmsEntry.Base) bar), 
-                e
-            );
+    public static class ShardTooLargeException extends RfsException {
+        public ShardTooLargeException(long shardSizeBytes, long maxShardSize) {
+            super("The shard size of " + shardSizeBytes + " bytes exceeds the maximum shard size of " + maxShardSize + " bytes");
         }
     }
 
-    @Override
-    public String getPhaseName() {
-        return "Documents Migration";
-    }
+    private void doDocumentsMigration(IndexAndShard indexAndShard) {
+        log.info("Migrating docs for " + indexAndShard);
+        ShardMetadata.Data shardMetadata = shardMetadataFactory.apply(indexAndShard.indexName, indexAndShard.shard);
 
-    @Override
-    public Logger getLogger() {
-        return logger;
-    }
+        try (var unpacker = unpackerFactory.create(shardMetadata)) {
+            var reader = readerFactory.apply(unpacker.unpack());
+            Flux<Document> documents = reader.readDocuments();
 
-    public static class DocumentsMigrationPhaseFailed extends Runner.PhaseFailed {
-        public DocumentsMigrationPhaseFailed(GlobalState.Phase phase, WorkerStep nextStep, Optional<CmsEntry.Base> cmsEntry, Exception e) {
-            super("Documents Migration Phase failed", phase, nextStep, cmsEntry, e);
+            final int finalShardId = shardMetadata.getShardId(); // Define in local context for the lambda
```

Comment:
I don't think this comment is accurate, no?

Reply:
You're right & I've shed the whole line & inlined it into a logging lambda anyway now.